### DISCUSSION TITLE
QIcon and browser search

### DISCUSSION THREAD

**Discussion by zefyx** [quasarframework/quasar#14339] [General - Components / Directives / etc] [answered]
Hello,
While using the integrated search of a browser (CTRL + F), i came across the fact that de QIcon (https://quasar.dev/vue-components/icon#introduction) component put the icon name inside the generated tag
`<q-icon name="today"/>`  become `<i class="q-icon notranslate material-icons" aria-hidden="true" role="presentation">today</i>` .
The word "today" , as the result, appear in the search results and the browser highlight an empty space on the icon (depending of the status of the part of the word found). 

If the QIcon is used inside another component often used with a loop (Qlist or Qtable for example), the word "today" is found as many times as the QIcon is renderer thus "drowning" the word expected to be found in text. 
The result is the same with a [QBtn ](https://quasar.dev/vue-components/button#introduction) and using the props "icon"

Below a search for "Ed" ( in the context of searching all the users starting with "Ed" in a list)

![image_327](https://user-images.githubusercontent.com/113043125/188923708-47417c64-52ba-48de-8cdb-520c1a50460a.png)

Using the available slot with a dummy content (QTooltip) or ``<q-icon name="edit">&nbsp;</q-icon>`` doesn't work either.

Any idea on how to "solve" or bypass this small problem ?

**Comment by pdanpdan** [comment, marked as answer]
This is the way material icon font works.
You can try to use material as SVG

**Comment by zefyx** [comment]
Hello @pdanpdan,

Thank you for your answer.
Coming from "Font Awesome", i didn't know that !

For people finding this topic : see [Official Google Material Icons Guide](https://developers.google.com/fonts/docs/material_icons#using_the_icons_in_html)
There is an old related discussion : https://github.com/google/material-design-icons/issues/498
The codepoints link mentioned is obsolete. See https://github.com/google/material-design-icons/blob/master/font/MaterialIcons-Regular.codepoints

In my case, and as suggested by @pdanpdan, i chose the SVG way ([QIcon component#Svg icons](https://quasar.dev/vue-components/icon#svg-icons))
